### PR TITLE
Improve Doc Generation

### PR DESCRIPTION
**Pull request by bbernays**
Ran into issues when changing a table name where the github checks kept failing because the old table file was never being deleted. By clearing out all of the files for each run this will avoid that issue

### CODE REVIEW

File: docs/docs.go [hunk 1]
```diff
@@ -9,7 +9,13 @@ import (
 )
 
 func main() {
-	err := docs.GenerateDocs(resources.Provider(), "./docs")
+	outputPath := "./docs"
+	err := os.RemoveAll(outputPath)
+	if err != nil {
+		fmt.Printf("%+v", err)
```
Comment (yevgenypats) [inline docs/docs.go:15]: Shouldn't it exit with non nil if err ?

Comment (yevgenypats) [reply docs/docs.go:15]: I think the next err is also missing `os.exit(1)`

**Review by yevgenypats — COMMENTED** [review]
looks good few minor comments

**Comment by roneli** [comment]
Would add this feature to the SDK as well so all providers can have this feature.

File: docs/docs.go (hunk 1 above)
Comment (bbernays) [reply docs/docs.go:15]: Agreed

**Comment by bbernays** [comment]
> Would add this feature to the SDK as well so all providers can have this feature.

I originally thought to just add it to the SDK but was concerned about adding destructive actions... wasn't sure if the directory was always 100% dynamically generated or if there could be manual additions

**Comment by roneli** [comment]
> > Would add this feature to the SDK as well so all providers can have this feature.
> 
> I originally thought to just add it to the SDK but was concerned about adding destructive actions... wasn't sure if the directory was always 100% dynamically generated or if there could be manual additions

We can add it as an optional flag `delete_old bool` to the doc generator. I see it useful for all other providers we support.

**Comment by bbernays** [comment]
@roneli - Created an issue on the cq-provider-sdk to implement this functionality

File: docs/docs.go [hunk 2]
```diff
@@ -2,14 +2,31 @@ package main
...
+			fmt.Fprintf(os.Stderr, "Failed to generate docs: %s\n", err)
+			os.Exit(1)
+		}
+
+	}
+
+	err = docs.GenerateDocs(resources.Provider(), outputPath)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Failed to generate docs: %s\n", err)
 	}
```
Comment (roneli) [inline docs/docs.go:32]:
```suggestion
	if err = docs.GenerateDocs(resources.Provider(), outputPath); err != nil {
		fmt.Fprintf(os.Stderr, "Failed to generate docs: %s\n", err)
	}
```

File: docs/docs.go [hunk 3]
```diff
@@ -2,14 +2,31 @@ package main
...
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to generate docs: %s\n", err)
+		os.Exit(1)
+	}
+	for _, d := range dir {
+		err = os.RemoveAll(path.Join([]string{outputPath + "/tables", d.Name()}...))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to generate docs: %s\n", err)
+			os.Exit(1)
+		}
```
Comment (roneli) [inline docs/docs.go:25]:
```suggestion
		if err := os.RemoveAll(path.Join([]string{outputPath + "/tables", d.Name()}...)); err != nil {
			fmt.Fprintf(os.Stderr, "Failed to generate docs: %s\n", err)
			os.Exit(1)
		}
```

**Review by roneli — APPROVED** [review]
LGTM, added two suggestions

**Pull request merged**